packages/fxa-auth-server/test/local/redis.js

```diff
@@ -4,140 +4,187 @@
-        result = await wrapper.foo();
-      } catch (e) {
-        err = e;
-      }
+      await redis.del(uid);
+      await redis.touchSessionToken(uid, sessionToken);
+      await redis.touchSessionToken(uid, { ...sessionToken, id: 'token2' });
     });
 
-    it('rejected with 409 conflict', () => {
-      assert.isUndefined(result);
-      assert.isObject(err);
-      assert.equal(err.errno, error.ERRNO.REDIS_CONFLICT);
-      assert.equal(err.message, 'Redis WATCH detected a conflicting update');
-      assert.equal(err.output.statusCode, 409);
-      assert.equal(err.output.payload.error, 'Conflict');
+    it('does nothing for unknown uids', async () => {
+      await redis.pruneSessionTokens('x');
+      const tokens = await redis.getSessionTokens('x');
+      assert.isEmpty(tokens);
     });
 
-    it('called the underlying redis method', () => {
-      assert.equal(redis.foo.callCount, 1);
+    it('does nothing for unkown token ids', async () => {
+      await redis.pruneSessionTokens(uid, ['x', 'y']);
+      const tokens = await redis.getSessionTokens(uid);
+      assert.deepEqual(Object.keys(tokens), [sessionToken.id, 'token2']);
     });
-  });
 
-  describe('failing method call:', () => {
-    let result, err;
+    it('deletes a given token id', async () => {
+      await redis.pruneSessionTokens(uid, ['token2']);
+      const tokens = await redis.getSessionTokens(uid);
+      assert.deepEqual(Object.keys(tokens), [sessionToken.id]);
+    });
 
-    beforeEach(async () => {
+    it('deleted the uid record when no tokens remain', async () => {
+      await redis.pruneSessionTokens(uid, [sessionToken.id, 'token2']);
+      const rawData = await redis.get(uid);
+      assert.isNull(rawData);
+    });
+  });
+});
+
+describe('Redis down', () => {
+  describe('touchSessionToken', () => {
+    it('returns without error', async () => {
       try {
-        mockError = new Error('wibble');
-        result = await wrapper.foo();
-      } catch (e) {
-        err = e;
+        await downRedis.touchSessionToken(uid, {});
+      } catch (err) {
+        assert.fail();
       }
     });
+  });
 
-    it('rejected with 500 error', () => {
-      assert.isUndefined(result);
-      assert.isObject(err);
-      assert.equal(err.errno, error.ERRNO.UNEXPECTED_ERROR);
-      assert.equal(err.message, 'Unspecified error');
-      assert.equal(err.output.statusCode, 500);
-      assert.equal(err.output.payload.error, 'Internal Server Error');
+  describe('getSessionTokens', () => {
+    it('returns an empty object without error', async () => {
+      const tokens = await downRedis.getSessionTokens(uid);
+      assert.isEmpty(tokens);
     });
+  });
 
-    it('called the underlying redis method', () => {
-      assert.equal(redis.foo.callCount, 1);
+  describe('pruneSessionTokens', () => {
+    it('throws a timeout error', async () => {
+      try {
+        await downRedis.pruneSessionTokens(uid);
+      } catch (e) {
```

Comment:
Could we use something like https://github.com/mozilla/fxa/blob/master/packages/fxa-auth-server/test/assert.js#L15 here?

Reply:
i suppose. for now i prefer this since it's one less level of indirection